### ISSUE TITLE
implement save and reload test for contrastive model

### ISSUE THREAD

**Issue by ebursztein** [tensorflow/similarity#181]
Model currently only implement save.

**Comment by Lunatik00** [comment]
if you are referring to the load of the model, i have tried saving an h5 file and i couldn't, i am searching about that now, for now the way to save an load a model would be 
```
--------------------------------------------
your code to generate the model
--------------------------------------------
model.save(directory_path)
```
and for load 

```
from tensorflow.keras.models import load_model
load_model(directory_path)
```
you also can save just the weights and load the weights only after making the model with a function

**Comment by owenvallis** [comment]
Added support for save and reload in ce00de50dfa985087a0805fcafa09d3e4953a742.

Users can now call `my_contrastive_model.save(save_path, save_index=True)`. This calls save() on each of the sub models, writes a json file containing the top level args required by the model and the serialized loss, optimizer, and metrics, saves the optimizer weights as a npy file, and optionally saves the index. 

We also added a `contrastive_model.load_model(save_path)` function that should recreate the model, init and set the optimizer weights, and supports training from the saved state of the model.
